src/main/git/gitTypes.ts

```diff
@@ -1,23 +1,23 @@
 export interface Repository {
-    name: string,
-    author: string,
-    url: string
+  name: string;
+  author: string;
+  url: string;
 }
 
 export interface File {
-    filename: string,
-    path: string
+  filename: string;
+  path: string;
 }
 
 export enum Url {
-    REDIRECT_URI = "http://localhost:3000/main_window",
-    AUTHORIZE_URL = "https://github.com/login/oauth/authorize",
-    ACCESS_TOKEN_URL = "https://github.com/login/oauth/access_token"
+  REDIRECT_URI = 'http://localhost:3000/main_window',
```

Comment:
I think this will crash if we build the app, because it would not be running on localhost:3000 anymore, am I right?  
Anyway, maybe we could just do a relative path redirect to `/main_window` instead? It is safer and won't crash if port ever changes

Reply:
I have not built this app ever before so honestly don't have any idea how it behaves in production. I have a feeling that since it's a desktop app Electron will work it's magic and the callback in production will still be localhost just not sure if the port will match. Because what sort of URL would it be if not a localhost.

Reply:
@lukasz-starosta using `http://localhost/main_window` works in development as well so I'm going to default to this for now. There is almost no info on the web about any sort of `root_url` for Electron apps or how to handle this and doing realtive `/main_window` does not work unfortunately as it would be relative to GitHub's url so this way at least we're getting rid of the port that could cause problems.

Based on my observations it will probably be `http://localhost/main_window` in production as well as the app runs locally and `main_window` is simply the app name and the point of entry is defined by webpack here:
```json
"plugins": [
  [
    "@electron-forge/plugin-webpack",
    {
      "mainConfig": "./webpack.main.config.js",
      "renderer": {
        "config": "./webpack.renderer.config.js",
        "entryPoints": [
          {
            "html": "./public/index.html",
            "js": "./src/renderer/renderer.tsx",
            "name": "main_window" <----
          }
        ]
      }
    }
  ]
]
```
and then used like this:
```ts
declare const MAIN_WINDOW_WEBPACK_ENTRY: any;
declare const MAIN_WINDOW_PRELOAD_WEBPACK_ENTRY: any;
.
.
.
const mainWindow = new BrowserWindow({
    height: 600,
    width: 1200,
    webPreferences: {
      preload: MAIN_WINDOW_PRELOAD_WEBPACK_ENTRY,
      enableRemoteModule: true,
      nodeIntegration: true,
    },
  });

  mainWindow.loadURL(MAIN_WINDOW_WEBPACK_ENTRY); <----
```

Definitely would be good to test the packaged up at some stage though as these are just assumptions.